### PR TITLE
mrc-1476 refactor report related models

### PR DESCRIPTION
**Pull request by hillalex**
Prior to factoring out a `ReportRepository` and `ReportLogic`, this refactors the `ReportVersion` models - this is so that the repo can return basic objects derived from the `REPORT` and `REPORT_VERSION` tables and then the logic class can compose the complex objects that require also fetching from other tables (`ARTEFACT`, `RESOURCE`, etc.)

### CODE REVIEW

File: src/app/src/main/kotlin/org/vaccineimpact/orderlyweb/db/Orderly.kt [hunk 1]
```diff
@@ -28,50 +22,43 @@ class Orderly(val isReviewer: Boolean,
...
+    override fun getGlobalPinnedReports(): List<Report>
     {
         JooqContext().use {
 
-            // create a temp table containing the latest visible version ID for each report name
-            val latestVersionForEachReport = getLatestVersionsForReports(it)
-
-            val versions =  it.dsl.withTemporaryTable(latestVersionForEachReport)
-                    .select(REPORT_VERSION.REPORT,
+            val versions = it.dsl
```
Comment (hillalex) [inline src/app/src/main/kotlin/org/vaccineimpact/orderlyweb/db/Orderly.kt:49]: Also noticed that for the pinned reports we were fetching these whole complex `ReportVersion` objects with aretfacts, custom fields etc when the only properties needed are the date, id and display name. So simplified this method and changed to return `Report` objects.

File: src/app/src/main/kotlin/org/vaccineimpact/orderlyweb/db/Orderly.kt [hunk 2]
```diff
@@ -224,29 +206,29 @@ class Orderly(val isReviewer: Boolean,
...
             val thisVersion = getReportVersion(name, version, it)
-            return getDatedChangelogForReport(thisVersion.report, thisVersion.date, it)
-
+            return getDatedChangelogForReport(thisVersion.name, Timestamp.from(thisVersion.date), it)
         }
 
     }
 
-    override fun checkVersionExistsForReport(name: String, version: String)
+    override fun getReportVersion(name: String, version: String): BasicReportVersion
```
Comment (hillalex) [inline src/app/src/main/kotlin/org/vaccineimpact/orderlyweb/db/Orderly.kt:214]: Renamed and changed to return the fetched object - this method will be moved to the report repo and used to get the basic version model when building up the more complex model in the logic class

File: src/app/src/main/kotlin/org/vaccineimpact/orderlyweb/models/Report.kt [hunk 3]
```diff
@@ -9,14 +9,26 @@ constructor(val name: String,
             val displayName: String?,
             val latestVersion: String)
 
-data class ReportVersion
-@ConstructorProperties("name", "displayname", "id", "latestVersion", "published", "date", "customFields")
+data class BasicReportVersion
```
[the remark this answers is not in the document]
Comment (hillalex) [reply src/app/src/main/kotlin/org/vaccineimpact/orderlyweb/models/Report.kt:12]: good shout yeah

[the remark this answers is not in the document]
Comment (hillalex) [reply src/app/src/main/kotlin/org/vaccineimpact/orderlyweb/models/Report.kt:12]: or do you think ReportVersionDetail should go in the same file as them too?

[the remark this answers is not in the document]
Comment (EmmaLRussell) [reply src/app/src/main/kotlin/org/vaccineimpact/orderlyweb/models/Report.kt:12]: Don't really mind, think it's ok separate, just worried that the other ReportVersion classes didn't really belong in the Report file.

**Pull request merged**